cpp/src/join/mixed_join_common_utils.cuh

```diff
@@ -31,7 +31,25 @@ namespace detail {
 
 using row_hash =
   cudf::experimental::row::hash::device_row_hasher<cudf::hashing::detail::default_hash,
-                                                   cudf::nullate::DYNAMIC>;
+                                                   cudf::nullate::DYNAMIC,
+                                                   cudf::experimental::type_identity>;
+
+using row_hash_no_nested = cudf::experimental::row::hash::device_row_hasher<
+  cudf::hashing::detail::default_hash,
+  cudf::nullate::DYNAMIC,
+  cudf::experimental::dispatch_void_conditional_generator<id_to_type<type_id::STRUCT>,
+                                                          id_to_type<type_id::LIST>>::type>;
+
+using row_hash_no_complex = cudf::experimental::row::hash::device_row_hasher<
+  cudf::hashing::detail::default_hash,
+  cudf::nullate::DYNAMIC,
+  cudf::experimental::dispatch_void_conditional_generator<id_to_type<type_id::STRUCT>,
+                                                          id_to_type<type_id::LIST>,
+                                                          id_to_type<type_id::DECIMAL128>,
+                                                          id_to_type<type_id::DECIMAL64>,
+                                                          id_to_type<type_id::DECIMAL32>,
```

Comment:
I don't think decimal types are complex type. They are just a wrapper around some integer type.

Reply:
Let me know if we could resolve this. I have addressed this here https://github.com/rapidsai/cudf/pull/15700#discussion_r1765039562